src/core/mesh/qgstopologicalmesh.cpp

```diff
@@ -1897,6 +1916,195 @@ QgsTopologicalMesh::Changes QgsTopologicalMesh::addVertexInface( int includingFa
   return changes;
 }
 
+QgsTopologicalMesh::Changes QgsTopologicalMesh::insertVertexInFacesEdge( int faceIndex, int position, const QgsMeshVertex &vertexToInsert )
+{
+  const QgsMeshFace face1 = mMesh->face( faceIndex );
+
+//  const int addedVertexIndex = mMesh->vertexCount();
+//  int newFaceIndexStartIndex = mMesh->faceCount();
+
+  Changes changes;
+  changes.mVerticesToAdd.append( vertexToInsert );
+  changes.mAddedFacesFirstIndex = mMesh->faceCount();
+  // triangulate the first face
+
+  int newVertexPositionInFace1 = position + 1;
+
+  auto triangulate = [this, &changes]( int removedFaceIndex, const QgsMeshVertex & newVertex, int newVertexPosition, QVector<int> &edgeFacesIndexes )->bool
```

Comment:
can we make new function (static?) out of it for readability?

Reply:
not sure, with a static function I could not access to the private member of Changes class